src/comments/comment.service.ts

```diff
@@ -0,0 +1,360 @@
+        },
+      };
+    }
+  }
+
+  async getCommentsById(commentId: number): Promise<CommentDto> {
+    const comment = await this.commentsRepository.findOneBy({ id: commentId });
+
+    if (!comment) {
+      throw new NotFoundException(`Comment with ID ${commentId} not found`);
+    }
+
+    const commentDto: CommentDto = {
+      id: comment.id,
+      commentableId: comment.commentableId,
+      commentableType: comment.commentableType,
+      content: comment.content,
+      user: comment.user,
+      created_at: comment.createdAt.getTime(),
+      agree_type: 0,
+      agree_count: 0,
+      disagree_count: 0,
+      quote: {
+        quote_id: null,
+        quote_user: null,
+      },
+    };
+
+    return commentDto;
+  }
+
+  async deleteComment(commentId: number): Promise<void> {
+    const comment = await this.commentsRepository.findOneBy({ id: commentId });
+
+    if (!comment) {
+      throw new NotFoundException(`Comment with id ${commentId} not found`);
+    }
+
+    await this.commentsRepository.remove(comment);
+  }
+
+  async commentAnswer(answerId: number, content: string): Promise<CommentDto> {
+    const answer = await this.answersRepository.findOneBy({ id: answerId });
+
+    if (!answer) {
+      throw new Error('Answer not found');
+    }
+
+    const comment = new Comment();
+    comment.content = content;
+    comment.commentableType = 'answer';
+    comment.commentableId = answer.id;
+
+    const createdComment = await this.commentsRepository.save(comment);
+
+    const commentDto: CommentDto = {
+      id: createdComment.id,
+      content: createdComment.content,
+      commentableId: comment.commentableId,
+      commentableType: comment.commentableType,
+      user: comment.user,
+      quote: {
+        quote_id: null,
+        quote_user: null,
+      },
+      agree_count: 0,
+      disagree_count: 0,
+      agree_type: 0,
+      created_at: comment.createdAt.getTime(),
+    };
+
+    return commentDto;
+  }
+
+  async commentQuestion(
+    questionId: number,
+    content: string,
+  ): Promise<CommentDto> {
+    const question = await this.questionsRepository.findOneBy({
+      id: questionId,
+    });
+
+    if (!question) {
+      throw new Error('Answer not found');
+    }
+
+    // 创建评论
+    const comment = new Comment();
+    comment.content = content;
+    comment.commentableType = 'question';
+    comment.commentableId = question.id;
+    const createdComment = await this.commentsRepository.save(comment);
+    const commentDto: CommentDto = {
+      id: createdComment.id,
+      content: createdComment.content,
+      commentableId: comment.commentableId,
+      commentableType: comment.commentableType,
+      user: comment.user,
+      quote: {
+        quote_id: null,
+        quote_user: null,
+      },
+      agree_count: 0,
+      disagree_count: 0,
+      agree_type: 0,
+      created_at: comment.createdAt.getDate(),
+    };
+    return commentDto;
+  }
+
+  async commentComment(
```

Comment:
This function is not exposed in controller. Consider add a API and implement it.

Reply:
The API definitions here need an overhaul. You should expose a `createComment` API with parameters `commentable_type` and `commentable_id` instead of `commentAnswer` and `commentComment` and probably in the future `commentQuestion` and `commentInsight` etc.

With regard to querying comments and sub_comments, it calls for some thinking. Utilizing JOIN cluase is the only sensible way in performance. See https://stackoverflow.com/questions/52785147/correct-database-schema-for-question-answer-system.